modules/synapse/default.nix

```diff
@@ -64,6 +65,8 @@ in
       '';
     };
 
+    enableSlidingSync = mkEnableOption "the experimental sliding sync proxy";
```

Comment:
This is a litle awkward since the sliding sync proxy requires specifying at least one secret. The synapse module doesn't expose a way to set it inside the module so you have to in addition to enabling sliding sync "automatically" specify it via a separate module:

```
services.matrix-next.sliding-sync.secretFile = "somepath";
```

Could just expose that option inside the synapse module maybe?

Alternatively the only thing the synapse module does is set defaults based on the synapse config rather than to whatever else. We could remove the entire option from the synapse module and still default to those synapse options in the sliding-sync module. Any users of the sliding sync module without the synapse module has to specify the options anyways.


Third alternative is to have another preStart script in the sliding-sync service that automatically generates the secret for you if the file doesn't exist. Stored as state on the system. Similar to what synapse (the program) does for its keys.

You cant change the key without wiping the database anyways so it is state just like the db. This would elliminate the need to specifiy anything via the sliding-sync module.

Reply:
Keeping it in the synapse module would also free the sliding-sync to default to the normal nixos's synapse config rather than this one